docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+    },
+    {
+      "name": "Product",
+      "type": "String"
+    },
+    {
+      "name": "units",
+      "type": "Integer"
+    }
+  ]
+}
+```
+Schema string is a JSON string defining a separator character for schema, quote character for fields with String types and fields along with their name, type and additional constraints if there are any.
+
+Following data endpoints are supported: 
+
+* **KafkaEnpoint**
+: To define Kafka Endpoint we need to specify Kafka broker ( host:port ), topic name and MessageFormat as seen in line 1 in above code.
+* **FileEndpoint**
+: It needs to be configured with directory path, file name and MessageFormat as in line 2 in above code 
+* **StreamEndpoint** 
+: This allows us to connect existing operator output or input to SQL query as a data source or sink respectively. StreamEndpoint needs immediate downstream operator's input port or immediate upstream operator's output port and field mapping for CSV data or POJO class. This will be explained in detail in next [example](#fusion-style-sql-application).
+
+
+### Using User Defined functions ( UDF ) in SQL query
+
+We can use our own scalar UDF, implemented in Java, in a SQL statement for data manipulation. But before the usage, we need to register the function with execution environment by using registerFunction API.
+
+```java
+sqlEnv = sqlEnv.registerFunction("APEXCONCAT", PureStyleSQLApplication.class, "apex_concat_str");
+```
+
+In above code, **_registerFunction_** takes UDF name to be used in SQL, JAVA class which implements static method and name of the same method as parameters. 
+Used static method "apex_concat_str" takes two String objects as input parameters as a requirement from SQL query.
+```java    
+public static String apex_concat_str(String s1, String s2)
+{
+  return s1 + s2;
+}
+```
+
+The scalar UDF "APEXCONCAT" can be used in SQL as described below whereas FLOOR & SUBSTRING are standard SQL scalar functions supported by Apache Calcite.
+```sql
+INSERT INTO SALES 
+       SELECT STREAM ROWTIME, FLOOR(ROWTIME TO DAY), APEXCONCAT('OILPAINT', SUBSTRING(PRODUCT, 6, 7)) 
+       FROM ORDERS 
+       WHERE ID > 3 AND PRODUCT LIKE 'paint%'
+```
+To read about all functions and operators supported by Apache Calcite, click [here](https://calcite.apache.org/docs/reference.html#operators-and-functions).
+
+### Executing SQL Query 
+Finally to execute the query we need to use **_executeSQL_** API with DAG and SQL statement as parameters.
+
+```java
+// Converting SQL statement to DAG 
+String sql = "INSERT INTO SALES 
+                SELECT STREAM ROWTIME, FLOOR(ROWTIME TO DAY), APEXCONCAT('OILPAINT', SUBSTRING(PRODUCT, 6, 7)) 
+                FROM ORDERS 
+                WHERE ID > 3 AND PRODUCT LIKE 'paint%'";
+sqlEnv.executeSQL(dag, sql);
+```
+When executeSQL API is called, specified SQL query go through various phases like conversion to relational algebra, optimization and planning in Calcite to generate Relation Expression Tree. 
```

Comment:
See note above regarding use of "API".
specified SQL query go through => the query goes through